File: src/main/java/com/manideepla/bookerang/HomeController.java
package com.manideepla.bookerang;


import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HomeController {

    @GetMapping("/home")
    public String home() {
//     TODO this config is not working for some reason, figure it out
        return "home";
    }
}

File: src/main/kotlin/com/manideepla/bookerang/HomeController.kt
package com.manideepla.bookerang

import jakarta.servlet.http.HttpSession
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping


@Controller
class HomeController {

    private val logger = LoggerFactory.getLogger(HomeController::class.java)

    @Autowired
    lateinit var httpSession: HttpSession

    @GetMapping("/")
    fun showHome(): String {
        val username = httpSession.getAttribute("username")

        logger.info("user: $username")

        return if (username != null) {
            "redirect:/user/home?username=${username}"
        } else {
            "redirect:/user/login"
        }
    }
}